### PR TITLE
explain: add index merge

### PR DESCRIPTION
**Pull request by Joyinqin**
### What is changed, added or deleted? (Required)
This is a followup PR to #3858. The content is taken from query-execution-plan#indexmerge-example. 

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/3924
- Other reference link(s): [理解 TiDB 执行计划](https://docs.pingcap.com/zh/tidb/stable/query-execution-plan#indexmerge-示例)

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Have version specific changes 
- [ ] Might cause conflicts

### CODE REVIEW

File: explain-index-merge.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+---
+title: 启用 `Explain` 语句的 `IndexMerge` 特性
```
Comment (TomShawn) [inline explain-index-merge.md:2]: Using ` backticks in the metadata section causes parsing failure. Avoid using it here.
```suggestion
title: 启用 Explain 语句的 IndexMerge 特性
```

Comment (Joyinqin) [reply explain-index-merge.md:2]: Got it

File: explain-index-merge.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
+---
+title: 启用 Explain 语句的 IndexMerge 特性
```
Comment (TomShawn) [inline explain-index-merge.md:2]:
```suggestion
title: 使用 EXPLAIN 查看 SQL 执行计划——启用 IndexMerge
```

File: explain-index-merge.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+summary: 了解 TiDB 中 `EXPLAIN` 语句返回的执行计划信息。
+---
+
+# 启用 `Explain` 语句的 `IndexMerge` 特性
```
Comment (TomShawn) [inline explain-index-merge.md:6]:
```suggestion
# 使用 EXPLAIN 查看 SQL 执行计划——启用 IndexMerge
```

File: explain-index-merge.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+比如，在上述示例中，过滤条件是使用 `OR` 条件连接的 `WHERE` 子句。在启用 `IndexMerge` 前，每个表只能使用一个索引，不能将 `a = 1` 下推到索引 `a`，也不能将 `b = 1` 下推到索引 `b`。当 `t` 中存在大量数据时，全表扫描的效率会很低。针对这类场景，TiDB 引入了对表的新访问方式 `IndexMerge`。
+
+在 `IndexMerge` 访问方式下，优化器可以选择对一张表使用多个索引，并将每个索引的返回结果进行合并，生成以上示例中后一个 `IndexMerge` 的执行计划。此时的 `IndexMerge_16` 算子有三个孩子节点，其中 `IndexRangeScan_13` 和 `IndexRangeScan_14` 根据范围扫描得到符合条件的所有 `RowID`，再由 `TableRowIDScan_15` 算子根据这些 `RowID` 精确地读取所有满足条件的数据。
+
+其中对于 `IndexRangeScan`/`TableRangeScan` 一类按范围进行的扫表操作，`EXPLAIN` 表中 `operator info` 列相比于其他扫表操作，多了被扫描数据的范围这一信息。比如上面的例子中，`IndexRangeScan_13` 算子中的 `range:(1,+inf]` 这一信息表示该算子扫描了从 1 到正无穷这个范围的数据。
+
+> **注意：**
+>
+> 目前，TiDB 的 `IndexMerge` 特性在 TiDB 4.0.0-rc.1 版本中默认关闭。同时 4.0 版本中的 `IndexMerge` 目前支持的场景仅限于析取范式（`or` 连接的表达式）。开启 `IndexMerge` 特性有以下方法：
```
Comment (TomShawn) [inline explain-index-merge.md:39]: @lzmhhh123 Please help confirm whether it is `4.0.0-rc.1` or `4.0-rc`, thanks!
```suggestion
> 目前，TiDB 的 `IndexMerge` 特性在 TiDB 4.0.0-rc.1 版本中默认关闭。同时 4.0 版本中的 `IndexMerge` 目前支持的场景仅限于析取范式（`or` 连接的表达式），暂不支持合取范式（`and` 连接的表达式）。开启 `IndexMerge` 特性有以下方法：
```

**Review by lzmhhh123 — COMMENTED** [review]
LGTM

File: explain-index-merge.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
+---
+title: 使用 EXPLAIN 查看 SQL 执行计划——启用 IndexMerge
```
Comment (TomShawn) [inline explain-index-merge.md:2]:
```suggestion
title: 用 EXPLAIN 查看开启 IndexMerge 的 SQL 执行计划
```

File: explain-index-merge.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+summary: 了解 TiDB 中 EXPLAIN 语句返回的执行计划信息。
+---
+
+# 使用 EXPLAIN 查看 SQL 执行计划——启用 IndexMerge
```
Comment (TomShawn) [inline explain-index-merge.md:6]:
```suggestion
# 用 EXPLAIN 查看开启 IndexMerge 的 SQL 执行计划
```

**Review by TomShawn — APPROVED** [review]
LGTM

**Comment by TomShawn** [comment]
follow-up: TOC

**Pull request merged**